src/ws/service/friendships_service.rs

```diff
@@ -130,3 +163,83 @@ impl FriendshipsServiceServer<SocialContext> for MyFriendshipsService {
         todo!()
     }
 }
+
+/// Retrieve the User Id associated with the given Authentication Token.
+///
+/// If an authentication token was provided in the request, this function gets the
+/// user id from the token and returns it as a `Result<UserId, FriendshipsError>`. If no
+/// authentication token was provided, this function returns a `FriendshipsError::Unauthorized`
+/// error.
+///
+/// * `requests` -
+/// * `context` -
+async fn get_user_id_from_request(
+    request: &Payload,
+    context: &Arc<SocialContext>,
+) -> Result<UserId, FriendshipsError> {
+    match request.synapse_token.clone() {
+        // If an authentication token was provided, get the user id from the token
+        Some(token) => get_user_id_from_token(context.app_components.clone(), &token)
+            .await
+            .map_err(FriendshipsError::CommonError),
+        // If no authentication token was provided, return an Unauthorized error.
+        None => {
+            log::debug!("Get Friends > Get User ID from Token > `synapse_token` is None.");
+            Err(FriendshipsError::CommonError(CommonError::Unauthorized))
```

Comment:
I'd prefer to avoid using this `CommonError` as it's part of the API V1 Rest module, can we use something specific for ws?